scripts/dev/tidy.php

```diff
@@ -0,0 +1,310 @@
+    '/^.+\.jpg$/',
+    '/^.+\.tiff$/',
+    '/^.+\.phar$/',
+    '/^.+\.bz2$/',
+    '/^.+\.zip$/',
+    '/^.+\.data$/',
+    '/^.+\.webp$/',
+    '/^.+\.bmp$/',
+    '/^.+\.ttf$/',
+    '/^.+\.gd2$/',
+    '/^.+\.gif$/',
+    '/^.+\.ico$/',
+    '/^.+\.mov$/',
+    '/^.+\.gz$/',
+    '/^.+\.patch$/',
+    '/^sapi\/[a-zA-Z0-9_-]+\/tests\/.+$/',
+    '/^ext\/[a-zA-Z0-9_-]+\/tests\/.+$/',
+];
+
+// Check system requirements
+if (!extension_loaded("fileinfo")) die("This script requires the PHP fileinfo extension.");
+
+/**
+ * Main fixer class
+ */
+class Fixer
+{
+    /**
+     * Determine the EOL (end of line) type based on prevailing LF, CRLF, NEL
+     * usage in file. For empty files empty string is returned.
+     */
+    public function eol(string $content): string
+    {
+        $finfo = new finfo(FILEINFO_CONTINUE);
+        $info = $finfo->buffer($content);
+
+        $newLine = '';
+        if (strpos($info, 'CRLF line terminators') !== false) {
+            $newLine = "\r\n";
+        } elseif (strpos($info, 'CRLF, NEL line terminators') !== false) {
+            $newLine = "\r\n";
+        } elseif (strpos($info, 'CRLF, LF line terminators') !== false) {
+            $newLine = "\n";
+        } elseif ($info != 'empty') {
+            $newLine = "\n";
+        }
+
+        return $newLine;
+    }
+
+    /**
+     * Trims all trailing whitespace in given content.
+     */
+    public function trimWhitespace(string $content): string
+    {
+        return preg_replace('/[\t ]+(\r?)$/m', '', $content);
+    }
+
+    /**
+     * Trims all final new lines from given file and append an EOL based on previous
+     * prevailing EOL type used in the given string.
+     */
+    public function trimFinalNewLine(string $content): string
+    {
+        $trimmed = rtrim($content);
+        $trimmed .= $this->eol($trimmed);
+
+        return $trimmed;
+    }
+}
+
+class Logger
+{
+    private $green="\e[32m";
+    private $red="\e[31m";
+    private $default="\e[39m";
+
+    /**
+     * Log message information to STDOUT.
+     */
+    public function info(string $file, string $message): string
+    {
+        return $file.': '.$this->green.$message.$this->default."\n";
+    }
+
+    /**
+     * Log error information to STDOUT.
+     */
+    public function error(string $file, string $message): string
+    {
+        return $file.': '.$this->red.$message.$this->default."\n";
+    }
+}
+
+// Process arguments
+if ($argc < 2) {
+    echo usage();
+    exit;
+}
+
+array_shift($argv);
+
+foreach($argv as $argument) {
+    switch($argument) {
+        case '-t':
+        case '--trim':
+            $trim = true;
+        break;
+        case '--no-whitespace':
+            $whitespace = false;
+        break;
+        case '--no-eol':
+            $eol = false;
+        break;
+        case '-h':
+        case '--help':
+            echo usage();
+            exit;
+        break;
+        case (preg_match('/^-.+/', $argument) ? true : false):
```

Comment:
Please separate these out of the switch. Or make the entire switch a switch(true).

Reply:
Current script now outputs the error if invalid option is given.